src/drivers/optical_flow/paa3905/PAA3905.cpp

```diff
@@ -415,19 +415,14 @@ uint8_t PAA3905::RegisterRead(uint8_t reg)
 		px4_udelay(TIME_TSWR_us - elapsed_last_write);
 	}
 
-	// tSRW/tSRR SPI Time Between Read And Subsequent Commands
-	const hrt_abstime elapsed_last_read = hrt_elapsed_time(&_last_read_time);
-
-	if (elapsed_last_write < TIME_TSRW_TSRR_us) {
```

Comment:
This was just wrong I guess? Should have been `elapsed_last_read`?

If that is fixed, is the change still necessary?

Reply:
If we go by the letter of the datasheet we should be respecting these times, but they're lower on newer models and we can get away with it.